core/server/worker/src/main/java/alluxio/worker/grpc/AbstractReadHandler.java

```diff
@@ -337,7 +408,7 @@ private void runInternal() {
 
           if (chunk != null) {
             DataBuffer finalChunk = chunk;
-            mSerializingExecutor.execute(() -> {
+            mSerializingExecutor.addTask(() -> {
```

Comment:
Why not send it in-line?

Reply:
Sorry, I don't quite understand

Reply:
Again, why do we need to spawn a new thread?

Reply:
This doesn't spawn a new thread, only sends a task to the thread responsible for sending messages back to the client